include/moqt/publish_track_handler.h

```diff
@@ -0,0 +1,267 @@
+/*
+ *  Copyright (C) 2024
+ *  Cisco Systems, Inc.
+ *  All Rights Reserved
+ */
+#pragma once
+
+#include <functional>
+#include <moqt/core/base_track_handler.h>
+#include <moqt/metrics.h>
+
+namespace moq::transport {
+
+    /**
+     * @brief MOQ track handler for published track
+     *
+     * @details MOQ publish track handler defines all track related callbacks and
+     *  functions for publish. Track handler operates on a single track (namespace + name).
+     *
+     *  This extends the base track handler to add publish (aka send) handling
+     */
+    class PublishTrackHandler : protected BaseTrackHandler
+    {
+      public:
+        friend class Transport;
+
+        /**
+         * @brief Publish status codes
+         */
+        enum class PublishStatus : uint8_t
+        {
+            kOk = 0,
+            kInternalError,
+            kNotAuthorized,
+            kNotAnnounced,
```

Comment:
above 2 might be better scoped to Track status

Reply:
There is overlap on error/status conditions depending on which method you call. In this case, `PublishObjectStatus` is returned, which due to the async nature, you may no longer be authorized, etc.